apis/placement/v1alpha1/override_types.go

```diff
@@ -66,6 +67,12 @@ type OverridePolicy struct {
 
 // OverrideRule defines how to override the selected resources on the target clusters.
 type OverrideRule struct {
+	// OverrideType defines the type of the override rules.
+	// +kubebuilder:validation:Enum=JSONPatch;Delete
+	// +kubebuilder:default:JSONPatch
+	// +optional
+	OverrideType OverrideType `json:"overrideType,omitempty"`
```

Comment:
Better to move this to be after ClusterSelector.

Why?
- Specify the target and then specify the desired action upon the target.
- OverrideType and JSONPatchOverrrides are used to describe the action. They should be together.